File: Assets/Bundles/Server/Resources/TS/Services/Global/Damage/Projectile/ProjectileCollideServerSignal.lua
-- Compiled with unity-ts v2.1.0-75
local ProjectileCollideServerSignal
do
	ProjectileCollideServerSignal = setmetatable({}, {
		__tostring = function()
			return "ProjectileCollideServerSignal"
		end,
	})
	ProjectileCollideServerSignal.__index = ProjectileCollideServerSignal
	function ProjectileCollideServerSignal.new(...)
		local self = setmetatable({}, ProjectileCollideServerSignal)
		return self:constructor(...) or self
	end
	function ProjectileCollideServerSignal:constructor(projectile, baseDamage, hitPosition, normal, velocity, hitEntity)
		self.projectile = projectile
		self.baseDamage = baseDamage
		self.hitPosition = hitPosition
		self.normal = normal
		self.velocity = velocity
		self.hitEntity = hitEntity
		self.damage = baseDamage
	end
end
return {
	ProjectileCollideServerSignal = ProjectileCollideServerSignal,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Server/Services/Damage/Projectile/ProjectileCollideServerSignal.ts
import { Entity } from "Shared/Entity/Entity";
import { Projectile } from "Shared/Projectile/Projectile";

export class ProjectileCollideServerSignal {
	public damage: number;

	constructor(
		public readonly projectile: Projectile,
		public readonly baseDamage: number,
		public readonly hitPosition: Vector3,
		public readonly normal: Vector3,
		public readonly velocity: Vector3,
		public readonly hitEntity: Entity | undefined,
	) {
		this.damage = baseDamage;
	}
}
